prog/heatmap.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package prog
```

Comment:
Please describe the general idea how the heatmap operates at the top of the file. And if you have ideas on how it could be improved / tailored to the specific underlying data type, post them also as comments.